flow-typed/civica/name-record.js

```diff
@@ -0,0 +1,20 @@
+import { type GeoscapeAddress } from '../geoscape'
+
+export type CivicaNameRecord = {
+  title: string,
+  givenName1: string,
+  familyName: string,
+  emailAddress?: string,
+  homePhone?: string,
+  businessPhone?: string,
+  mobilePhone?: string,
+  faxPhone?: string,
+  streetAddress: [
+    | {
+        address1: string,
+        address2: string,
+        postcode: string,
+      }
+    | GeoscapeAddress
```

Comment:
IMO, we do the mapping client side. So that the submission data is always a valid NR to be posted to Civica.